pkg/services/oauthtoken/oauth_token.go

```diff
@@ -52,20 +53,32 @@ func ProvideService(socialService social.Service, authInfoService login.AuthInfo
 }
 
 // GetCurrentOAuthToken returns the OAuth token, if any, for the authenticated user. Will try to refresh the token if it has expired.
-func (o *Service) GetCurrentOAuthToken(ctx context.Context, usr *user.SignedInUser) *oauth2.Token {
-	if usr == nil {
+func (o *Service) GetCurrentOAuthToken(ctx context.Context, usr identity.Requester) *oauth2.Token {
+	if usr == nil || usr.IsNil() {
 		// No user, therefore no token
 		return nil
 	}
 
-	authInfoQuery := &login.GetAuthInfoQuery{UserId: usr.UserID}
+	namespace, id := usr.GetNamespacedID()
+	if namespace != identity.NamespaceUser && namespace != identity.NamespaceRenderService {
```

Comment:
Is this always safe for render service namespace?

Reply:
render namespace can be user ID 0, I'll change it